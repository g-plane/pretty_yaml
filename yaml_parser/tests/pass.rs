use insta::{assert_snapshot, glob, Settings};
use std::{fs, path::Path};

#[test]
fn pass_snapshot() {
    glob!("pass/*.yaml", |path| {
        let input = fs::read_to_string(path).unwrap();
        let tree = match yaml_parser::parse(&input) {
            Ok(tree) => tree,
            Err(err) => panic!("failed to parse '{}': {err}", path.display()),
        };
        assert_eq!(
            tree.to_string(),
            input,
            "syntax tree of '{}' does not match source",
            path.display()
        );

        build_settings(path).bind(|| {
            let name = path.file_stem().unwrap().to_str().unwrap();
            assert_snapshot!(name, format!("{tree:#?}"));
        });
    });
}

fn build_settings(path: &Path) -> Settings {
    let mut settings = Settings::clone_current();
    settings.set_snapshot_path(path.parent().unwrap());
    settings.remove_snapshot_suffix();
    settings.set_prepend_module_to_snapshot(false);
    settings.remove_input_file();
    settings.set_omit_expression(true);
    settings.remove_input_file();
    settings.remove_info();
    settings
}
