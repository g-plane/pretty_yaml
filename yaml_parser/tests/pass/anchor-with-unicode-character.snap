---
source: yaml_parser/tests/pass.rs
---
ROOT@0..27
  DOCUMENT@0..26
    DIRECTIVES_END@0..3 "---"
    WHITESPACE@3..4 "\n"
    BLOCK@4..26
      BLOCK_SEQ@4..26
        BLOCK_SEQ_ENTRY@4..26
          MINUS@4..5 "-"
          WHITESPACE@5..6 " "
          BLOCK@6..26
            FLOW@6..26
              PROPERTIES@6..11
                ANCHOR_PROPERTY@6..11
                  AMPERSAND@6..7 "&"
                  ANCHOR_NAME@7..11 "üòÅ"
              WHITESPACE@11..12 " "
              PLAIN_SCALAR@12..26 "unicode anchor"
  WHITESPACE@26..27 "\n"
